ps_mbo.php

```diff
@@ -381,6 +406,24 @@ private function registerShop()
         }
     }
 
+    /**
+     * Unregister a shop of online services delivered by API.
+     * When the module is disabled or uninstalled, remove it from online services
+     *
+     * @return void
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    private function unregisterShop(): void
+    {
+        $token = $this->getAdminAuthenticationProvider()->getAdminToken();
+
+        /** @var Client $distributionApi */
+        $distributionApi = $this->getService('mbo.cdc.client.distribution_api');
+
+        $distributionApi->unregisterShop($token);
```

Comment:
Missing errors handling here
IMHO it shouldn't block the module uninstall/disable and we cannot retry it since the module is disabled/uninstalled